movieReviewApp/src/test/java/com/mobiquity/movieReviewApp/service/SignUpServiceImplTest.java

```diff
@@ -2,63 +2,58 @@
 import com.github.database.rider.junit5.api.DBRider;
-import com.mobiquity.movieReviewApp.model.ResetPassword;
-import com.mobiquity.movieReviewApp.model.UserProfile;
-import java.time.LocalDateTime;
+import com.mobiquity.movieReviewApp.exception.UserException;
+import com.mobiquity.movieReviewApp.model.UserInformation;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.extension.ExtendWith;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
-import org.springframework.test.context.ActiveProfiles;
-import org.springframework.test.context.junit.jupiter.SpringExtension;
 
-//@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+
 @SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
-//@DBRider
-//@DBUnit(leakHunter = true)
-@ActiveProfiles("unit-test")
+@DBRider
 class SignUpServiceImplTest {
 
   @Autowired
   SignUpService signUpService;
 
-
-  private String token = "eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJzQGdtYWlsLmNvbSAyIiwiZXhwIjoxNTgwOTA2NDg2LCJpYXQiOjE1ODA4MjAwODZ9.1EAgHyfosb4f7N9zIlPDlxiZBIekEgq5ZWHu0eCQft-B5Nd2C9gyAuiVO-dJ98f5JcLkHtG2W8cVDm2R_mf7KQ";
+  private UserInformation userInformation = new UserInformation();
 
   @Test
   public void checkIfEmailIdIsNotAlreadyRegistered() {
-    String result = signUpService.saveUser(
-        new UserProfile(1l,"movie", "s@gmail.com", "asdfg", false, LocalDateTime.now(),
-            LocalDateTime.now()));
+    userInformation.setPassword("asdfg");
+    userInformation.setName("movie");
+    userInformation.setEmailId("asdfgh@gmail.com");
+    String result = signUpService.saveUser(userInformation);
     assertEquals("Activate your link", result);
   }
 
   @Test
+  @DataSet(value = "data.xml")
   public void checkIfEmailIdIsAlreadyRegistered() {
-    String result = signUpService.saveUser(
-        new UserProfile(2l, "movie", "s@gmail.com", "asdertc", false, LocalDateTime.now(),
-            LocalDateTime.now()));
-    assertEquals("Your email is already registered.", result);
+    userInformation.setPassword("asdfg");
+    userInformation.setName("movie");
+    userInformation.setEmailId("www@gmail.com");
+    assertThrows(UserException.class, () -> signUpService.saveUser(
+        userInformation));
   }
 
   @Test
   public void checkIfTokenIsExpired() {
-    String result = signUpService.registerAccount(token);
-    assertEquals("Your activation link got expired", result);
+    String token = "eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJzQGdtYWlsLmNvbSAyIiwiZXhwIjoxNTgwOTA2NDg2LCJpYXQiOjE1ODA4MjAwODZ9.1EAgHyfosb4f7N9zIlPDlxiZBIekEgq5ZWHu0eCQft-B5Nd2C9gyAuiVO-dJ98f5JcLkHtG2W8cVDm2R_mf7KQ";
+    assertThrows(UserException.class, () -> signUpService.registerAccount(token));
+
```

Comment:
Does this work if you have: 
`Exception thrown = assertThrows(UserException.class, () -> signUpService.registerAccount(token));
    assertEquals("Your activation link got expired", thrown.getLocalizedMessage());`?

Reply:
It works as you suggested as well. But i feel assertThrows is more generic way to test whether the code throws exception or not.

Reply:
It depends.. some statements you are catching the User exception twice and giving it different messages . In that case you should be checking that the correct message is being thrown (or creating separate exceptions)